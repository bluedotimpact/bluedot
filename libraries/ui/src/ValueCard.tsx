import React from 'react';

interface ValueCardProps {
  icon: string;
  title: string;
  description: string;
}

export const ValueCard: React.FC<ValueCardProps> = ({ icon, title, description }) => {
  return (
    <div className="value-card container-lined w-[437px] h-[296px] p-6 flex flex-col gap-[27px]">
      <div className="value-card__icon-wrapper size-[50px] flex items-center justify-center">
        <img
          src={icon}
          alt={title}
          className="value-card__icon"
        />
      </div>
      <h3 className="value-card__title">
        {title}
      </h3>
      <p className="value-card__description">
        {description}
      </p>
    </div>
  );
};

export default ValueCard;
