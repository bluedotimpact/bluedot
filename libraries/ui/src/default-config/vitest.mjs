/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
