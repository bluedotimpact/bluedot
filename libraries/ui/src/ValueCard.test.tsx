import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { ValueCard } from './ValueCard';

describe('ValueCard', () => {
  test('renders default as expected', () => {
    const { container } = render(
      <ValueCard
        icon="/icons/test-icon.svg"
        title="Test Title"
        description="Test description"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with all props correctly', () => {
    const { container } = render(
      <ValueCard
        icon="/icons/custom-icon.svg"
        title="Custom Title"
        description="Custom description text that is longer"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
