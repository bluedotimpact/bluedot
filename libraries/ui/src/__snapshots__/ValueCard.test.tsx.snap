// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ValueCard > renders default as expected 1`] = `
<div>
  <div
    class="value-card w-full flex flex-col gap-6 py-6"
  >
    <div
      class="value-card__icon-wrapper size-[50px] flex items-center justify-center"
    >
      <img
        alt="Test Title"
        class="value-card__icon"
        src="/icons/test-icon.svg"
      />
    </div>
    <h3
      class="value-card__title min-h-[60px]"
    >
      Test Title
    </h3>
    <p
      class="value-card__description "
    >
      Test description
    </p>
  </div>
</div>
`;

exports[`ValueCard > renders with all props correctly 1`] = `
<div>
  <div
    class="value-card w-full flex flex-col gap-6 py-6"
  >
    <div
      class="value-card__icon-wrapper size-[50px] flex items-center justify-center"
    >
      <img
        alt="Custom Title"
        class="value-card__icon"
        src="/icons/custom-icon.svg"
      />
    </div>
    <h3
      class="value-card__title min-h-[60px]"
    >
      Custom Title
    </h3>
    <p
      class="value-card__description "
    >
      Custom description text that is longer
    </p>
  </div>
</div>
`;
